contracts/Stage1/TokenManager.sol

```diff
@@ -1,58 +1,64 @@
 //SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.8.0;
+pragma solidity ^0.8.15;
 
 import "@openzeppelin/contracts/access/Ownable.sol";
+import "contracts/interfaces/ISeuro.sol";
 
 contract TokenManager is Ownable {
+    string[] public tokenSymbols;
+    mapping(string => TokenData) tokenMetaData;
 
-    bytes32 private constant WETH_NAME = bytes32("WETH");
-    uint8 private constant WETH_DEC = 18;
-
-    Token[] public tokens;
-
-    struct Token { bytes32 name; address addr; uint8 dec; address chainlinkAddr; uint8 chainlinkDec; }
+    struct TokenData { address addr; uint8 dec; address chainlinkAddr; uint8 chainlinkDec; }
 
     /// @param _wethAddress address of WETH token
     /// @param _ethUsdCL address of Chainlink data feed for ETH / USD
     /// @param _ethUsdCLDec number of decimals that ETH / USD data feed uses
     constructor(address _wethAddress, address _ethUsdCL, uint8 _ethUsdCLDec) {
-        addDefaultTokens(_wethAddress, _ethUsdCL, _ethUsdCLDec);
+        addAcceptedToken(_wethAddress, _ethUsdCL, _ethUsdCLDec);
     }
 
     // Gets the details for the given token, if it is accepted
-    /// @param _name 32-byte array value representation of the token symbol e.g. "WETH", "USDT"
-    function get(bytes32 _name) external view returns(Token memory token) {
-        for (uint256 i = 0; i < tokens.length; i++) if (tokens[i].name == _name) token = tokens[i];
-        require(token.name != bytes32(0), "err-tok-not-found");
-    }
-
-    function addDefaultTokens(address _wethAddress, address _ethUsdCL, uint8 _ethUsdCLDec) private {
-        addAcceptedToken(WETH_NAME, _wethAddress, WETH_DEC, _ethUsdCL, _ethUsdCLDec);
+    /// @param _symbol The token symbol e.g. "WETH", "USDC", "USDT"
+    function get(string memory _symbol) external view returns(TokenData memory) {
+        for (uint256 i = 0; i < tokenSymbols.length; i++) if (cmpString(tokenSymbols[i], _symbol)) return tokenMetaData[_symbol];
+        revert("err-tok-not-found");
     }
 
-    // Get an array of all the accepted tokens
-    function getAcceptedTokens() external view returns (Token[] memory) {
-        return tokens;
+    function getAcceptedTokens() external view returns (string[] memory) {
+        return tokenSymbols;
```

Comment:
hm, i changed the storage of these tokens recently because it made more sense for the frontend to make a single call, and get all the addresses, decimals etc. rather than get the names then have to iterate and get the addresses, decimals for each token afterwards.

this would mean bradley would have to make n more calls to the token manager contract for however many tokens there are

Reply:
i created new api with getters for all these metadata. instead of calling it once and then parsing it in the frontend, query the blockchain directly. simple, only need to remember the symbol name.